From 681fb0770c04975c6c3276c46f10eb977217479d Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:20:38 -0400
Subject: [fs] nfsd: Remove nfs4_lock_state(): nfsd4_lock/locku/lockt()

Message-id: <1410898849-22126-155-git-send-email-bfields@redhat.com>
Patchwork-id: 95828
O-Subject: [Patch RHEL7 154/165] nfsd: Remove nfs4_lock_state(): nfsd4_lock/locku/lockt()
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 2dd7f2ad4e34a3487e071ad3ef88b0a1ae7a97f2
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 0f9e842..2712cd3 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -5205,8 +5205,6 @@ nfsd4_lock(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
   return status;
  }
 
- nfs4_lock_state();
-
  if (lock->lk_is_new) {
   if (nfsd4_has_session(cstate))
    /* See rfc 5661 18.10.3: given clientid is ignored: */
@@ -5349,7 +5347,6 @@ out:
  if (open_stp)
   nfs4_put_stid(&open_stp->st_stid);
  nfsd4_bump_seqid(cstate, status);
- nfs4_unlock_state();
  if (file_lock)
   locks_free_lock(file_lock);
  if (conflock)
@@ -5392,8 +5389,6 @@ nfsd4_lockt(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  if (check_lock_length(lockt->lt_offset, lockt->lt_length))
    return nfserr_inval;
 
- nfs4_lock_state();
-
  if (!nfsd4_has_session(cstate)) {
   status = lookup_clientid(&lockt->lt_clientid, cstate, nn);
   if (status)
@@ -5448,7 +5443,6 @@ nfsd4_lockt(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
 out:
  if (lo)
   nfs4_put_stateowner(&lo->lo_owner);
- nfs4_unlock_state();
  if (file_lock)
   locks_free_lock(file_lock);
  return status;
@@ -5472,8 +5466,6 @@ nfsd4_locku(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  if (check_lock_length(locku->lu_offset, locku->lu_length))
    return nfserr_inval;
 
- nfs4_lock_state();
-                 
  status = nfs4_preprocess_seqid_op(cstate, locku->lu_seqid,
      &locku->lu_stateid, NFS4_LOCK_STID,
      &stp, nn);
@@ -5516,7 +5508,6 @@ put_stateid:
  nfs4_put_stid(&stp->st_stid);
 out:
  nfsd4_bump_seqid(cstate, status);
- nfs4_unlock_state();
  if (file_lock)
   locks_free_lock(file_lock);
  return status;
-- 
1.7.1